.github/workflows/test.yml

```diff
@@ -305,7 +305,7 @@ jobs:
           name: "${{ github.sha }}-ab"
         if: "env.GIT_DIFF != ''"
       - name: Run tests with race detector
-        run: cat xab.txt | xargs go test -mod=readonly -json -timeout 15m -race -tags='cgo ledger test_ledger_mock' > xab-race-output.txt
+        run: cat xab.txt | xargs go test -mod=readonly -json -timeout 30m -race -tags='cgo ledger test_ledger_mock' > xab-race-output.txt
```

Comment:
@alessio I'm doubling the timeout here, because:
- it often fails on master https://github.com/cosmos/cosmos-sdk/commits/master
- when it doesn't fail, it finishes in ~14m, e.g. https://github.com/cosmos/cosmos-sdk/pull/7475/checks?check_run_id=1221058280

Reply:
There seems to be a test or something else that is consuming a massive amount time. Increasing the timeout to 30m defeats the purpose of splitting tests into multiple jobs (decrease overall time)

Reply:
Agreed 👍 . I see this more as a band-aid patch, as it's blocking ~3-4 automerge PRs. Unless someone has an idea for a quick fix?

Reply:
okay, would be good to try. If its hanging then 30m wont change it

Reply:
my hypothesis is that it's not hanging, it passes in some PRs, the test I linked above finished in 14m39s.

but anyways, "Cancelled after 15m" https://github.com/cosmos/cosmos-sdk/pull/7442/checks?check_run_id=1225173890#step:7:1 :(

Reply:
it's probably due to the IBC tests. We rely heavily on simapp since it's kinda hard to unit test interactions between different chains and using simapp makes testing a lot more robust. I noticed recently the test times on my machine almost doubled in duration, so I wonder if there was a change to simapp that slowed things down?

Reply:
the `-race` flag slows things down a lot. This plus the IBC tests could be the culprit.